src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,100 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert a phone " + phone
+                    + "in a DB!", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
```

Comment:
the idea of criteria query is in the code flexibility, it should be easy to extend current code base
try it like this:
```java
Predicate predicate = criteriaBuilder.conjunction(); // work as empty `and()` bellow
	    for (Map.Entry<String, String[]> entry : params.entrySet()) {
                predicate = criteriaBuilder.and(predicate, root.get(entry.getKey()).in(entry.getValue()));
            }
           ....
```